components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -76,24 +76,22 @@ class P3Microphysics : public AtmosphereProcess
     KOKKOS_INLINE_FUNCTION
     void operator()(const int icol) const {
       for (int ipack=0;ipack<m_npack;ipack++) {
-        // The ipack slice of input variables
-        const Spack pmid_pack(pmid(icol,ipack));
-        const Spack T_atm_pack(T_atm(icol,ipack));
-        const Spack pseudo_density_pack(pseudo_density(icol,ipack));
-        const Spack qv_pack(qv(icol,ipack));
-        const Spack cld_frac_t_pack(cld_frac_t(icol,ipack));
+        // The ipack slice of input variables used more than once
+        const Spack& pmid_pack(pmid(icol,ipack));
+        const Spack& T_atm_pack(T_atm(icol,ipack));
+        const Spack& cld_frac_t_pack(cld_frac_t(icol,ipack));
         // Exner
         const auto& exner = PF::exner_function(pmid_pack);
         inv_exner(icol,ipack) = 1.0/exner;
         // Potential temperature
         th_atm(icol,ipack) = PF::calculate_theta_from_T(T_atm_pack,pmid_pack);
         // DZ
-        dz(icol,ipack) = PF::calculate_dz(pseudo_density_pack, pmid_pack, T_atm_pack, qv_pack);
+        dz(icol,ipack) = PF::calculate_dz(pseudo_density(icol,ipack), pmid_pack, T_atm_pack, qv(icol,ipack));
         // Cloud fraction
         // Set minimum cloud fraction - avoids division by zero
-        cld_frac_l(icol,ipack) = ekat::max(cld_frac_t(icol,ipack),mincld);
-        cld_frac_i(icol,ipack) = ekat::max(cld_frac_t(icol,ipack),mincld);
-        cld_frac_r(icol,ipack) = ekat::max(cld_frac_t(icol,ipack),mincld);
+        cld_frac_l(icol,ipack) = ekat::max(cld_frac_t_pack,mincld);
+        cld_frac_i(icol,ipack) = ekat::max(cld_frac_t_pack,mincld);
+        cld_frac_r(icol,ipack) = ekat::max(cld_frac_t_pack,mincld);
```

Comment:
Since the rhs is the same for all three, you could use `ekat::max` only in the first:
```
auto& frac_l = cld_frac_i(icol,ipack);
frac_l = ekat::max(...);
cld_frac_i(icol,ipack) = frac_i;
cld_frac_r(icol,ipack) = frac_i;
```